server.py

```diff
@@ -0,0 +1,182 @@
+import grpc
+from concurrent import futures
+from google.protobuf import json_format
+import time
+import json
+import datetime
+import os
+
+#import the generated class
+from proto import user_pb2
+from proto import user_pb2_grpc
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy import create_engine
+
+#import the database connection
+import psycopg2
+from controller.user_controller import UserController
+from model.user_model import User
+from controller.functions import PBtimeToPtime, PtimeToPBtime
+
+from dateutil.relativedelta import relativedelta
+
+from protoc_gen_validate.validator import validate, ValidationFailed
+
+host=os.environ.get('POSTGRES_HOST')
+database= os.environ.get('POSTGRES_DB_NAME')
+user=os.environ.get('POSTGRES_USER')
+password= os.environ.get('POSTGRES_PASSWORD')
+
+db_link = 'postgresql://' + user + ':' + password + '@' + host + '/' + database
+
+engine = create_engine(db_link)
+Session = sessionmaker(bind=engine)
+session = Session()
+
+
+user_controller = UserController(session)
+
+
+class UserServicer(user_pb2_grpc.UserServicer):
+
+    def CreateUser(self, request, context):
+        response = user_pb2.CreateUserResponse()
+        version = int(1)
+
+        try:
+            validate(request)
+        except ValidationFailed as err:
+            response.status = str(err)
+            return response
+    
+        try:
+            #age verification
+        
+            if user_controller.user_exist(request.email) is not None:
+                response.status = "Email already used to register"
+
+            elif user_controller.user_name_available(request.user_name) is not None:
+                response.status = "user name taken!"
+
+            elif relativedelta(datetime.datetime.now(),PBtimeToPtime(request.date_of_birth)).years < int(18):
+                response.status = "You have to be over 18 years to register"
+                
+            else:   
+                user_controller.add_user(User(request.id, request.first_name, request.last_name, request.user_name, request.email, request.password, request.phone_number, PBtimeToPtime(request.date_of_birth), request.state_id, version))
```

Comment:
you need to check if `add_user` was successful before sending `success` to the user

Reply:
This version is not updated.

In the final update. This command was in a try block. So if there was any error in the common, it gets thrown to the exception block and the error is logged.